src/menu/menu.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { FilterQuery, Model } from 'mongoose';
+import { MealType } from '../shared/enums/meal-type.enum';
+import { CreateMenuDto } from './dto/create-menu.dto';
+import { UpdateMenuDto } from './dto/update-menu.dto';
+import { Menu, MenuDocument } from './schemas/menu.schema';
+
+@Injectable()
+export class MenuService {
+  constructor(@InjectModel(Menu.name) private menuModel: Model<MenuDocument>) {}
+
+  async create(createMenuDto: CreateMenuDto) {
+    const menu = new this.menuModel(createMenuDto);
+    return menu.save();
+  }
+
+  async findAll() {
+    return this.menuModel.find().exec();
+  }
+
+  async findOne(options: FilterQuery<MenuDocument>) {
+    return this.menuModel.findOne(options).exec();
+  }
+
+  async findByName(name: string) {
+    return this.menuModel.findOne({ name });
```

Comment:
You could simply use `this.findOne({ name })` instead of this way. Then you will have a simple abstraction and only one function to modify if there is some need to do it.